pym/bob/input.py

```diff
@@ -469,61 +469,17 @@ def onFinish(self, env, tools, properties, package):
         """
         pass
 
-class ScmOverride:
-    def __init__(self, override):
-        self.__match = override.get("match", {})
-        self.__del = override.get("del", [])
-        self.__set = override.get("set", {})
-        self.__replaceRaw = override.get("replace", {})
-        self.__init()
-
-    def __init(self):
-        self.__replace = { key : (re.compile(subst["pattern"]), subst["replacement"])
-            for (key, subst) in self.__replaceRaw.items() }
-
-    def __getstate__(self):
-        return (self.__match, self.__del, self.__set, self.__replaceRaw)
-
-    def __setstate__(self, s):
-        (self.__match, self.__del, self.__set, self.__replaceRaw) = s
-        self.__init()
-
-    def __doesMatch(self, scm):
-        for (key, value) in self.__match.items():
-            if key not in scm: return False
-            if not fnmatch.fnmatchcase(scm[key], value): return False
-        return True
-
-    def mangle(self, scm):
-        if self.__doesMatch(scm):
-            scm = scm.copy()
-            for d in self.__del:
-                if d in scm: del scm[d]
-            scm.update(self.__set)
-            for (key, (pat, repl)) in self.__replace.items():
-                if key in scm:
-                    scm[key] = re.sub(pat, repl, scm[key])
-        return scm
-
 def Scm(spec, env, overrides):
-    # resolve with environment
-    spec = { k : ( env.substitute(v, "checkoutSCM::"+k) if isinstance(v, str) else v)
-        for (k, v) in spec.items() }
-
-    # apply overrides
-    for override in overrides:
-        spec = override.mangle(spec)
```

Comment:
The scm mangling should stay here and should not be moved into the scms. Currently it is possible to change the scm type with a scmOverride. This could be used to rewrite a git checkout to an URL download for example.